File: src/api/controllers/health.ts
import { registrySize } from '../../mcp/registry/index.js';
import { auditLogStatus } from '../../core/audit/status.js';

export type HealthResponse = {
  status: 'ok';
  registrySize: number;
  lastAuditEvent?: string;
};

export const getHealthStatus = async (): Promise<HealthResponse> => {
  const [size, audit] = await Promise.all([registrySize(), auditLogStatus()]);

  return {
    status: 'ok',
    registrySize: size,
    lastAuditEvent: audit?.lastEventTime
  };
};

File: tornado_ai/api/controllers/health.py
"""Health endpoint helpers."""
from __future__ import annotations

from dataclasses import dataclass

from ...core.audit.status import audit_log_status
from ...mcp.registry import registry_size


@dataclass
class HealthResponse:
    status: str
    registrySize: int
    lastAuditEvent: str | None = None


async def get_health_status() -> HealthResponse:
    size = registry_size()
    audit = audit_log_status()
    return HealthResponse(
        status="ok",
        registrySize=size,
        lastAuditEvent=audit.last_event_time,
    )
